webrender/src/device/gfx/device.rs

```diff
@@ -3158,12 +2996,55 @@ impl<B: hal::Backend> Device<B> {
         self.frame_id.0 += 1;
     }
 
+    pub fn begin_render_pass(&mut self) {
+        assert!(!self.inside_renderpass);
+        assert_eq!(self.draw_target_usage, DrawTargetUsage::Draw);
+        let (frame_buffer, format, has_depth) = if self.bound_draw_fbo != DEFAULT_DRAW_FBO {
+            let rbo_id = self.fbos[&self.bound_draw_fbo].rbo;
+            (
+                &self.fbos[&self.bound_draw_fbo].fbo,
+                self.fbos[&self.bound_draw_fbo].format,
+                self.rbos.get(&rbo_id).is_some(),
+            )
+        } else {
+            let (frame_buffer, has_depth) = match self.current_depth_test {
+                None => (&self.framebuffers[self.current_frame_id], false),
+                _ => (&self.framebuffers_depth[self.current_frame_id], true),
+            };
+            (
+                frame_buffer,
+                self.surface_format,
+                has_depth,
+            )
+        };
+
+        let render_pass = self.render_passes.get_render_pass(format, has_depth);
+        unsafe {
+            self.command_buffer.begin_render_pass(
+                render_pass,
+                frame_buffer,
+                self.viewport.rect,
+                &[],
```

Comment:
we are never clearing anything at the start of a pass? I suspect this is temporary?

Reply:
Yes this is temporary.